### ISSUE TITLE
performAsynchronousMessageExchangeWithBlock:responseBlock: causes memory leak

### ISSUE THREAD

**Issue by jeremywhuff** [TheAmazingAudioEngine/TheAmazingAudioEngine#125]
I have a class that uses this function, and objects of my class were sometimes never deallocated. I traced the problem to this function.

If I called

[theAudioController performAsynchronousMessageExchangeWithBlock:^{
     // Some code here
     } responseBlock:nil];

then the object would never get deallocated. However, if I instead called

[theAudioController performAsynchronousMessageExchangeWithBlock:^{
     // Some code here
     } responseBlock:^{}];

then the memory management would work properly, and the object would get deallocated.

So to summarize, sending nil as the response block causes an extra retain on the sender.
